source/fundamentals/object-serialization.txt

```diff
@@ -0,0 +1,202 @@
+   MongoCollection<BasicDBObject> basicDBObjectCollection =
+       database.getCollection("exampleCollection", BasicDBObject.class);
+   MongoCollection<BsonDocument> BsonDocumentCollection =
+       database.getCollection("exampleCollection", BsonDocument.class);
+
+Custom Codecs
+-------------
+
+You can represent data in a ``MongoCollection`` using Java classes for
+which serialization is not supported by default with a **custom codec**.
+
+Create a Custom Codec
+~~~~~~~~~~~~~~~~~~~~~
+
+You can create a custom codec for a class in the following ways:
+
+- Use an existing non-default codec that is compatible
+  with your class, such as :java-docs:`UuidCodec
+  <apidocs/bson/org/bson/codecs/UuidCodec.html>`.
+
+- Create your own codec by implementing the :java-docs:`Codec
+  <apidocs/bson/org/bson/codecs/Codec.html>` interface in a
+  codec class that serializes and deserializes your class.
+
+Use a Custom Codec
+~~~~~~~~~~~~~~~~~~
+
+To use a custom codec to represent data in a ``MongoCollection``, add
+your custom codec to a :java-docs:`CodecRegistry
+<apidocs/bson/org/bson/codecs/configuration/CodecRegistry.html>`
+and configure the ``CodecRegistry`` containing your codec as the codec
+provider for your ``MongoClient``, ``MongoDatabase``, or
+``MongoCollection``.
+
+.. note::
+
+   ``MongoDatabase`` instances inherit their
+   ``CodecRegistry`` from the ``MongoClient`` used to create them.
+   ``MongoCollection`` instances inherit their ``CodecRegistry`` from
+   the ``MongoDatabase`` used to create them. However, you can create
+   a new instance of a ``MongoDatabase`` or ``MongoCollection`` with
+   an alternate ``CodecRegistry`` using the ``withCodecRegistry()``
+   method.
+
+Use a Custom Codec with a Client
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To customize the ``CodecRegistry`` of a ``MongoClient``, pass a
+``CodecRegistry`` to the :java-docs:`MongoClientSettings.Builder.codecRegistry()
+<apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#codecRegistry(org.bson.codecs.configuration.CodecRegistry)>` method in the
+``MongoClientSettings`` builder object and pass the
+:java-api-sync:`MongoClientSettings <com/mongodb/MongoClientSettings.html>`
+generated by the ``build()`` method to the ``MongoClients.create()``
+method.
+
+The following example creates a ``MongoClient`` with a
+``CodecRegistry`` containing the default codecs for ``Document``,
+``BasicDBObject``, and ``BsonDocument`` as well as a custom codec of
+type ``ExampleCodec``:
+
+.. code-block:: java
+
+   CodecRegistry codecRegistry =
+      CodecRegistries.fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+                CodecRegistries.fromCodecs(new ExampleCodec()));
+   MongoClientSettings settings = MongoClientSettings.builder()
+                .codecRegistry(codecRegistry).build();
+   MongoClient client = MongoClients.create(settings);
+
+All ``MongoDatabase`` objects created from the ``MongoClient``
```

Comment:
That doesn't have the same meaning; the point is that any time you create a MongoDatabase, it automatically gets the CodecRegistry of the MongoClient that created it (unless specifically overridden). It's not just custom registries that inherit, and I think complicating the inheritance rules would only confuse our readers.

Reply:
Hm, not sure I see why what I've written doesn't convey that. I did intentionally skip bridging inheritance of the registry from Database->Collection since I thought it would help simplify the explanation, but maybe it's not great to imply that. I am more focused on the adhering to the clear and concise sentences from the Style Guide.

Also not sure what you mean that "not just custom registries that inherit"